pulp_service/pulp_service/app/authorization.py

```diff
@@ -23,33 +26,52 @@ def has_permission(self, request, view):
         # Admins have all permissions
         if request.user.is_superuser:
             return True
-        # Decode the identity header
+        user = request.user
+
+        # The user that created the domain has permission to access that domain
+        if DomainOrg.objects.filter(domain=get_domain(), user=user).exists():
+            return True
+
+        # Check if user is creating a domain or creating a resource within a domain
+        action = self.get_user_action(request)
+
+        # Decode the Red Hat Identity header
+        decoded_header_content = self.get_decoded_identity_header(request)
+
+        # Get the Org ID from the Red Hat Identity header
+        org_id = self.get_org_id(decoded_header_content)
+
+        # Anyone can create a domain
+        if action == "domain_create":
+            if decoded_header_content:
+                org_id_var.set(org_id)
+            user_id_var.set(request.user.pk)
+            return True
+        # User has permission if the org_id matches the domain's org_id
+        else:
+            return DomainOrg.objects.filter(domain=get_domain(), org_id=org_id).exists()
+
+    def get_user_action(self, request):
+        if request.META['PATH_INFO'].endswith('/default/api/v3/domains/') and request.META['REQUEST_METHOD'] == 'POST':
```

Comment:
What about the case where it's a non-domain operation, e.g. listing a worker? I think that would mistakenly be marked here as `domain_operation`?

Reply:
Yes, it is a domain operation. You are listing workers within the default domain. No one except the admin has the right to use the default domain. We can give certain users access to the default domain if we decide to.